Lib/tempfile.py

```diff
@@ -747,28 +757,43 @@ def name(self):
     def newlines(self):
         return self._file.newlines
 
+    def readable(self):
+        return self._file.readable()
+
     def read(self, *args):
         return self._file.read(*args)
 
+    def read1(self, *args):
+        return self._file.read1(*args)
+
+    def readinto(self, b):
+        return self._file.readinto(b)
+
+    def readinto1(self, b):
+        return self._file.readinto1(b)
+
     def readline(self, *args):
         return self._file.readline(*args)
 
     def readlines(self, *args):
         return self._file.readlines(*args)
 
+    def seekable(self):
+        return self._file.seekable()
+
     def seek(self, *args):
         return self._file.seek(*args)
 
     def tell(self):
         return self._file.tell()
 
     def truncate(self, size=None):
-        if size is None:
-            self._file.truncate()
-        else:
-            if size > self._max_size:
-                self.rollover()
-            self._file.truncate(size)
+        if size is not None and size > self._max_size:
+            self.rollover()
+        return self._file.truncate(size)
```

Comment:
No need to change code that is working! 🙂

Reply:
I figured that a small change to the flow would be fine since I was changing the function anyway to make it return the result of the `self._file.truncate()` call. I can leave it mostly the same and just add the 2 required `return`s if you really want, but I'd prefer to keep it how it is now as I think it makes the intended logic more clear.

Reply:
The cosmetic change made me miss the real change!
I would prefer having two returns added and the rest of the structure unchanged.
This keeps commit info for each line accurate (therefore useful later).